bgpd/bgp_routemap.c

```diff
@@ -1441,7 +1441,7 @@ route_set_evpn_gateway_ip(void *rule, const struct prefix *prefix, void *object)
 
 	/* Set gateway-ip value. */
 	bre->type = OVERLAY_INDEX_GATEWAY_IP;
-	memcpy(&bre->gw_ip, &gw_ip->ip.addr, IPADDRSZ(gw_ip));
+	memcpy(&bre->gw_ip, &gw_ip, IPADDRSZ(gw_ip));
```

Comment:
isn't `IPADDRSZ` still going to cause a problem?

Reply:
Oof, good catch, thanks! it will indeed miss `ipa_type`-sized memory from the copy, truncating the address.

Fixed!